### DISCUSSION TITLE
Default protocol for Deskflow

### DISCUSSION THREAD

**Discussion by nbolton** [deskflow/deskflow#7742] [Polls]
Please let us know which protocol you think we should use by default in Deskflow.

The Synergy and Barrier protocols are identical except for the _hello_ message; one is not intrinsically better than the other. However, there are effects of choosing one or the other as the default.

If we make the Synergy protocol the default in Deskflow:
- Existing Synergy and Deskflow clients will work automatically
- Input Leap and Barrier clients **_will not work_**  by default (a setting must be changed)

If we make the Barrier protocol the default in Deskflow:
- Input Leap and Barrier clients will work automatically
- Existing Synergy and Deskflow clients **_will not work_** by default (a setting must be changed)

There are millions of Synergy client installs out in the wild (open source). There are only a few Deskflow installs out in the wild as it's so new. Barrier and Input Leap numbers are unknown.

**Comment by p12tic** [comment]
Maintainer of Input Leap here.

I wouldn't recommend defaulting on the same protocol as Barrier/Input Leap. The protocol uses hello string with application name and version number not just because, but in attempt to make protocol upgrades possible. If Deskflow uses identical protocol it couples Input Leap and Deskflow together and makes protocol upgrades way harder. Consider the following cases:
 1. Input Leap wants to do a 1.7 protocol revision and Deskflow wants to do a different 1.7 protocol revision or thinks that Input Leap protocol revision is stupid and does not want to support it.
 2. Input Leap fixes a bug in current 1.6 protocol and that fix causes a crash in Deskflow.
 3. Deskflow fixes a bug in current 1.6 protocol and that fix causes a crash in Input Leap.
 
 I can't force Deskflow not to support Input Leap protocol. However the possibility of Deskflow causing crashes in Input Leap means that Input Leap will need to not trust the other side, parse messages defensively, attempt detection of communication with other clients and warn the user that this is not supported.

**Reply by sithlord48** [reply]
More of a reason we should have a single proto lib that we both use

**Reply by nbolton** [reply]
@p12tic Thanks for contributing to the conversation. Glad to have you  here.

We _could_ bifurcate the protocols if we wanted to, but we don't have to. Wouldn't it be more collaborative and in the spirit of open source to use a shared library between the two projects?

Let's work together! :)

**Comment by nbolton** [comment]
![image](https://github.com/user-attachments/assets/2b2b97a9-8914-471c-a092-afdc79b1a3ba)

Final results are in: Barrier protocol by default wins (by a hair's breadth). 

If that's what the people want then let's give it to them.
